### PR TITLE
Remove SID as default owner of installer/installer

### PR DESCRIPTION
**Pull request by mads-hartmann**
## Description

In the interest of reducing unnecessary required PR reviews form SID I'd like to remove us as the default owners of `install/installer`

A recent example PR is https://github.com/gitpod-io/gitpod/pull/16633 where `install/installer/pkg/common` was changed. There are no explicit CODEOWNERS for  that folder, so i falls back to the owner of `install/installer` which is us. 

We could add an entry of `install/installer/pkg/common <blank>` to CODEOWNERS to remove us, but I'd rather remove us as the default owner so we don't have to list ever current (and future) folder in install/installer where we have shared ownership.

## Related Issue(s)

N/A

## How to test

N/A

## Release Notes

```release-note
NONE
```

## Build Options:

- [ ] /werft with-github-actions
      Experimental feature to run the build with GitHub Actions (and not in Werft).
- [ ] leeway-no-cache
      leeway-target=components:all
- [ ] /werft no-test
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer Options</summary>

- [ ] with-ee-license
- [ ] with-slow-database
- [ ] with-dedicated-emulation
- [ ] with-ws-manager-mk2
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -30,7 +30,6 @@
 /components/installation-telemetry @gitpod-io/engineering-security-infrastructure-and-delivery
 /components/kots-config-check @gitpod-io/engineering-security-infrastructure-and-delivery
 /install @gitpod-io/engineering-security-infrastructure-and-delivery
-/install/installer @gitpod-io/engineering-security-infrastructure-and-delivery
```
Comment (corneliusludmann) [inline .github/CODEOWNERS:33]: Wouldn't removing this line leads to a fallback to the SID team again due to the following line?
```
/install @gitpod-io/engineering-security-infrastructure-and-delivery
```

I think we need this (empty owner) instead, don't we? :thinking: 
```
/install/installer
```

**Review by corneliusludmann — CHANGES_REQUESTED** [review]
In general, I would love to divide the ownership of the installer into proper parts for each team (split it into different files etc.). However, in favor of embracing the chaos and focus on the important tasks, I agree that this is the proper way right now.

Added a comment below. Otherwise: Go for it and thanks for bias to action! :pray:

File: .github/CODEOWNERS (hunk 1 above)
Comment (mads-hartmann) [reply .github/CODEOWNERS:33]: You're right, I had missed the above line. Let me update the PR ☺️

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -30,11 +30,8 @@
...
-# For testdata a single review from anyone who is allowed to review PRs is sufficent.
-/install/installer/cmd/testdata
-# Allow everyone to approve config changes
-/install/installer/pgk/config
```
Comment (mads-hartmann) [inline .github/CODEOWNERS:37]: We no longer need to have these folders as explicitly "shared" anymore so I removed them

**Review by corneliusludmann — APPROVED** [review]

**Pull request merged**